codegen/src/test/java/org/web3j/codegen/SolidityFunctionWrapperTest.java

```diff
@@ -902,6 +902,133 @@ public void testBuildFuncNameConstants() throws Exception {
         assertEquals(builder.build().toString(), (expected));
     }
 
+    @Test
+    public void testBuildFunctionDuplicatedEventNames() throws Exception {
+
+        AbiDefinition firstEventDefinition =
+                new AbiDefinition(
+                        false,
+                        Arrays.asList(
+                                new NamedType("action", "string", false),
+                                new NamedType("pauseState", "bool", false)),
+                        "eventName",
+                        Collections.emptyList(),
+                        "event",
+                        false);
+        AbiDefinition secondEventDefinition =
+                new AbiDefinition(
+                        false,
+                        Arrays.asList(
+                                new NamedType("cToken", "address", false),
+                                new NamedType("action", "string", false),
+                                new NamedType("pauseState", "bool", false)),
+                        "eventName",
+                        Collections.emptyList(),
+                        "event",
+                        false);
+        TypeSpec.Builder builder = TypeSpec.classBuilder("testClass");
+        builder.addMethods(
+                solidityFunctionWrapper.buildFunctionDefinitions(
+                        "testClass",
+                        builder,
+                        Arrays.asList(firstEventDefinition, secondEventDefinition)));
+
+        String expected =
+                "class testClass {\n"
+                        + "  public static final org.web3j.abi.datatypes.Event EVENTNAME_EVENT = new org.web3j.abi.datatypes.Event(\"eventName\", \n"
+                        + "      java.util.Arrays.<org.web3j.abi.TypeReference<?>>asList(new org.web3j.abi.TypeReference<org.web3j.abi.datatypes.Utf8String>() {}, new org.web3j.abi.TypeReference<org.web3j.abi.datatypes.Bool>() {}));\n"
+                        + "  ;\n"
+                        + "\n"
+                        + "  public static final org.web3j.abi.datatypes.Event EVENTNAME1_EVENT = new org.web3j.abi.datatypes.Event(\"eventName1\", \n"
```

Comment:
Event name shouldn't be modified  `new org.web3j.abi.datatypes.Event(\"eventName1\"...` will fail in request.

Reply:
Can you please give me some more details about that?
Do you mean it is not possible to use the abiDef.setName() function on TYPE_EVENTS objects?

Thanks